src/modality-specific-files/eye-tracking.md

```diff
@@ -0,0 +1,213 @@
+# Eye-tracking including gaze position and pupil size
+
+Support for eye-tracking dataset developed as a
+[BIDS Extension Proposal](../extensions.md#bids-extension-proposals).
+Please see [Citing BIDS](../introduction.md#citing-bids) on how to
+appropriately credit this extension when referring to it in the context
+of the academic literature.
+
+## Terminology and conventions
+
+Eyetracking-BIDS is fully consistent with the BIDS specification as a whole.
+However, BIDS was initially developed in the context of MRI,
+so some terminology may be unfamiliar to researchers from other fields.
+This section adds clarifications to
+[Common Principles - Definitions](../common-principles.md#definitions) for the
+eye-tracking context.
+
+-   **Eye-tracker** - refers to the apparatus allowing the records of gaze
+    position and/or pupil size.
+
+-   **Screen** - Video display used to present visual stimulation (for example
+    projector, monitor, tablet).
+
+## Eye-tracking data
+
+With EyeTracking-BIDS, we wish to promote the adoption of good practices in
+the management of scientific data.
+
+### Template
+
+<!--
+This block generates a filetree exanple.
+A guide for editing it can be found at
+ https://github.com/bids-standard/bids-specification/blob/master/macros_doc.md.
+-->
+{{ MACROS___make_filetree_example(
+
+   {
+   "sub-<label>": {
+      "<datatype>": {
+         "<matches>_eyetrack.json": "",
+         "<matches>_eyetrack.tsv[.gz]": "",
+         },
+      }
+   }
+
+) }}
+
+Eye-tracking data MUST be stored in the main data recording modality or
+ `<datatype>` directory (for example `func` when combined with fMRI, or
+ `beh` when combined with behavioral measures).
+
+In the template filenames, the `<matches>` part corresponds to task filename
+before the suffix. For example, when fMRI is considered the main modality with
+files such as `sub-01_ses-1_task-pRF_run-1_bold.nii.gz`, `<matches>` would
+then correspond to `sub-01_ses-1_task-pRF_run-1`.
+
+### Tabular file (`*_eyetrack.tsv[.gz]`)
+
+The eye-tracking data must be saved in `.tsv` or `.tsv.gz` format.
+The tabular files consist of one row per sample and a set of REQUIRED and
+OPTIONAL columns.
+
+<!--
+This block generates a columns table.
+The definitions of these fields can be found in
+  src/schema/rules/tabular_data/eyetrack.yaml
+and a guide for using macros can be found at
+ https://github.com/bids-standard/bids-specification/blob/master/macros_doc.md
+-->
+{{ MACROS___make_columns_table("eyetrack.EyeTracking") }}
+
+Throughout BIDS you can indicate missing values with `n/a` (for "not
+available").
+
+We encourage users to put the eye-tracking raw data within the
+`/sourcedata` directory (for example put eye-tracker proprietary files before
+conversion to `tsv.gz` format in the sourcedata directory).
+
```

Comment:
The way we are interpreting this BEP would be yes, you *can* do that if you define those fields in the sidecar JSON. However, it could be useful to list those fields as optional and provide an idea for the naming.

Reply:
In the case of this BEP, we are considering saccades and fixations as derivatives (you need a model and parameters to get these values which are processed over the raw data), and blinks are encoded as loss of data. 
Eyetracker company provide sometimes these values with or without the model behind but it wasn't the goal of this BEP. I personally believe it could/shoud be a future BIDSapp goal to provide these info.

Reply:
No, not anything produced by a measurement require a model. 
A saccade is a particular response, it is not directly observable, it needs formulas, methods, and parameters to be detected, it is then the output of the measurement the eyetracker is producing. In this way saccade are derivatives. Importantly, not every eye tracker provide them and the provided parameters of the saccade and fixation are dependent on the parameters chosen by the eyetracker company. So clearly it is not the goal of the BEP as long as we aim here at defining a format to share raw data of eyetracking.

Reply:
> No, not anything produced by a measurement require a model.

This is not the point of my comment, nor it relates to what BIDS states it should cover.

> So clearly it is not the goal of the BEP as long as we aim here at defining a format to share raw data of eyetracking.

This is just your opinion, which I do not share. And if that is the goal of the BEP (which I don't think it is), then (as I said above) the goal should change and include *anything* that is produced by *any* eye-tracking device, attempting to be as inclusive as preparing the BEP allows.

Reply:
> Hi again, I see this has started some conversation! 
The main reason I asked is that I'm helping to prepare a eyelink dataset for publication. All our analysis is based on the saccades and fixations defined by eyelink software, so I believe this information should be shared with our data. Since no other indication was given in the bids proposal, my thought was to include these as derivatives.
Given the current bids proposal, all information from the raw data is shared in a standardized format (all the 'header' info is included in the _eyetracking.json file and relevant task event messages are in the _events.* files ), so I think having a standard way to include saccade and fixation would be very beneficial.

Thanks for your message, your rationale was good, it can be saved in derivatives/ folder of BIDS structure. 
We are discussing below of a way to save trigger messages, maybe this will be a option to those who want to save these results in the BIDS folder. I will keep you updated.

Reply:
Couple of "big picture" comments:

1. Goal of the BEP

>  the goal [of the BEP] should change and include *anything* that is produced by *any* eye-tracking device, attempting to be as inclusive as preparing the BEP allows

- by not tackling saccades, blinks... there was also a bit of the 80/20 principles at play and trying to focus on the most common use case shared by all devices.
- not everything produced by any device should have a name in BIDS: many json sidecars of MRI data contain very extensive dumps of the DICOM metadata, yet only a handful of those are accepted terms in BIDS

2. devrived VS raw

The last "derivative meeting" tried to help operationalize what is meant by derivative dataset in BIDS:

> If a dataset is derived from at least one other valid BIDS dataset, then it is a derivative dataset.

[source](https://bids-specification.readthedocs.io/en/latest/derivatives/introduction.html#bids-derivatives)

So I would have to say that from that point of view saccades and blinks outputed by an eyetracker could be considered as raw (even if they were generated using some unknown model).

In the world of BIDS MRI there is at least one precedent where the same file can end in different places based on this: [MP2RAGE UNI files](https://bids-specification.readthedocs.io/en/latest/appendices/qmri.html#unit1-images).

> Although the UNIT1 image is provided as an output by the acquisition sequence, it is used as an input to offline calculation of a T1map using a dictionary lookup approach. However, complex data is needed for an accurate calculation of the UNIT1 image, which is not commonly provided by the stock sequence. Instead, the magnitude and phase images are exported. Please see the relevant discussion at [qMRLab issue #255](https://github.com/qMRLab/qMRLab/issues/255).
>
> **Therefore, the UNIT1 image provided by the scanner is RECOMMENDED to be stored under the anat raw dataset directory along with the MP2RAGE file collection and to be used as the primary input for quantifying a T1map.**
>
> **If an additional UNIT1 image is calculated offline, then the output is to be stored in the derivatives directory with necessary provenance information.**

Reply:
I suggest in the discussion to stick to the current specification with regards to [raw versus derivative](https://bids-specification.readthedocs.io/en/stable/derivatives/introduction.html#derivatives-storage-and-directory-structure) datasets:

> Raw data are data that have been curated into BIDS from a non-BIDS source. If a dataset is derived from at least one other valid BIDS dataset, then it is a derivative dataset.

So although online or offline saccade detection requires some model (just like going from k-space to MR images requires a model), it is the data curation process that determines whether something is a derivative. If the saccades do not stem from a raw BIDS dataset but from an EDF or ASC file, then they can be considered for inclusion in the raw data. If an offline saccade detection algorithm was used on a raw BIDS eye tracking dataset, then the output of that algorithm would be considered a derivative.

Reply:
Circling back to @Lucrezia1011's question:

> Hi, are there any plans to include fixations, saccades and blinks defined by eyetracker software in a tabular format?

In our latest iteration, we have added the concept of "Physiology events". These are stored in `tsv.gz` files and MUST have an "onset" column (e.g., timestamps).

The new examples here: https://bids-specification--1128.org.readthedocs.build/en/1128/modality-specific-files/physiological-and-other-continuous-recordings.html#eye-tracking should show an example of fixations, saccades and blinks in tabular format.

Please note that you MAY use that file if (and only if) the eye-tracker provided those parameters. If those are the result of the researcher running a model fed with eye-tracking data, then these should go into derivatives.

Does that respond to the question?

Reply:
I'll mark this resolved -- @Lucrezia1011 please reopen if the latest messages did not really clarify this issue.